app/models/cart.rb

```diff
@@ -10,4 +10,10 @@ def add_product(product)
     end
     current_item
   end
+
+  #Contributes to being able to calculate total price of items in cart.
+  def total_price
+    line_items.to_a.sum { |item| item.total_price }
```

Comment:
The book doesn't always take the "most Ruby" approach to things, and this is one of those times. This can be simplified to:

```suggestion
    line_items.sum(&:total_price)
```

with the same effect. Depending on how much you wanna dig into this there is a StackOverflow post that goes into a bit more context [available here](https://stackoverflow.com/questions/17206254/difference-between-passing-method-and-method-as-function-arguments-in-ruby). But if you're not interested in beefing up on Ruby the way the book is going through this is more than enough.